lsr_role2collection.py

```diff
@@ -389,6 +394,56 @@ class LSRFileTransformer(LSRFileTransformerBase):
     """Do the role file transforms - fix role names, add FQCN
     to module names, etc."""
 
+    def convert_rolename(self, rolename, lsr_rolename=None):
+        """convert the given rolename to the new name"""
+        if rolename.count(".") == 1:
+            _src_owner, _rolename_base = rolename.split(".")
+        else:
+            _src_owner = None
+            _rolename_base = rolename
+        if not lsr_rolename:
+            lsr_rolename = self.src_owner + "." + self.rolename
+        logging.debug(f"\ttask role {rolename}")
+        new_name = None
+        if rolename == self.rolename or rolename == lsr_rolename:
```

Comment:
The previous code did this when being called from `task_cb`:
```
            if rolename == self.rolename or rolename == lsr_rolename:
```
but this when being called from `change_roles`:
```
                if role[key] == lsr_rolename or self.comp_rolenames(
                    role[key], self.rolename
                ):
```
Do you know why the comparison is different in the `task_cb` case and the `change_roles` case?  Why is `comp_rolenames` used?

Reply:
not sure - but I submitted another PR against your branch that uses `comp_rolenames` and cleans up some logic